src/core/include/openvino/op/eye.hpp

```diff
@@ -55,6 +55,11 @@ class OPENVINO_API Eye : public Op {
         m_output_type = output_type;
     }
 
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    bool evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const override;
```

Comment:
Update to TensorVector should be considered

Reply:
However, is it impossible to use `ov::Tensor` instead?

Reply:
I would say support for `ov::Tensor` in CPU shape_infer is a separate topic. 
Until `HostTensor` is used by shape_infer there, I think it is better to rely on compatible approach. 
It is possible to convert `ov::Tensor` to `HostTensor`, but such conversion will be done automatically by `ov::Model` or `ov::Node` if needed.